src/fragments/lib/datastore/flutter/data-access/delete-snippet.mdx

```diff
@@ -1,13 +1,18 @@
 Below, we query for an instance with an `id` of `123`, and then delete it, if found:
 
 ```dart
-(await Amplify.DataStore.query(Post.classType, where: Post.ID.eq('123')))
-    .forEach((element) async {
-  try {
-    await Amplify.DataStore.delete(element);
-    print('Deleted a post');
-  } on DataStoreException catch (e) {
-    print('Delete failed: $e');
+Future<void> deletePostsWithId() async {
+  final oldPosts = await Amplify.DataStore.query(
+    Post.classType,
+    where: Post.ID.eq('123'),
+  );
+  for (final oldPost in oldPosts) {
+    try {
+      await Amplify.DataStore.delete(oldPost);
```

Comment:
nit: The example query predicate ensures return 1 element with the result list, maybe unnecessary to demo a for loop here.

Reply:
Since one future relies on the other one and also for the sake of explaining better, I would rather keep the code as it is but improve the readability with comment and stating what @HuiSF mentioned